cmd/server.go

```diff
@@ -22,63 +22,67 @@ const (
-		ctx := context.Background()
-		ctx, cancel := context.WithCancel(ctx)
-		defer cancel()
+var (
+	port string
+	// serverCmd represents the server command
+	serverCmd = &cobra.Command{
+		Use:   "server",
+		Short: "Run PBnJ server",
+		Long:  `Run PBnJ server for interacting with BMCs.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			ctx := context.Background()
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
 
-		logger, zlog, err := logr.NewPacketLogr(
-			logr.WithServiceName("github.com/tinkerbell/pbnj"),
-			logr.WithLogLevel(logLevel),
-		)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "%v\n", err)
-			os.Exit(1)
-		}
-		defer zlog.Sync() // nolint
+			logger, zlog, err := logr.NewPacketLogr(
+				logr.WithServiceName("github.com/tinkerbell/pbnj"),
+				logr.WithLogLevel(logLevel),
+			)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "%v\n", err)
+				os.Exit(1)
+			}
+			defer zlog.Sync() // nolint
 
-		// Make sure that log statements internal to gRPC library are logged using the zapLogger as well.
-		grpc_zap.ReplaceGrpcLoggerV2(zlog)
+			// Make sure that log statements internal to gRPC library are logged using the zapLogger as well.
+			grpc_zap.ReplaceGrpcLoggerV2(zlog)
 
-		grpcServer := grpc.NewServer(
-			grpc_middleware.WithUnaryServerChain(
-				grpc_ctxtags.UnaryServerInterceptor(grpc_ctxtags.WithFieldExtractor(grpc_ctxtags.CodeGenRequestFieldExtractor)),
-				middleware.UnaryRequestID(middleware.UseXRequestIDMetadataOption(true), middleware.XRequestMetadataLimitOption(512)),
-				zaplog.UnaryLogRequestID(zlog, requestIDKey, requestIDLogKey),
-				grpc_zap.UnaryServerInterceptor(zlog),
-				grpc_validator.UnaryServerInterceptor(),
-			),
-		)
+			grpcServer := grpc.NewServer(
+				grpc_middleware.WithUnaryServerChain(
+					grpc_ctxtags.UnaryServerInterceptor(grpc_ctxtags.WithFieldExtractor(grpc_ctxtags.CodeGenRequestFieldExtractor)),
+					middleware.UnaryRequestID(middleware.UseXRequestIDMetadataOption(true), middleware.XRequestMetadataLimitOption(512)),
+					zaplog.UnaryLogRequestID(zlog, requestIDKey, requestIDLogKey),
+					grpc_zap.UnaryServerInterceptor(zlog),
+					grpc_validator.UnaryServerInterceptor(),
+				),
+			)
 
-		// reference for using another persistence layer
-		/*
-			opts := consul.Options{
-				Scheme:  "http",
-				Address: "localhost:8500",
-				Codec:   encoding.JSON,
-			}
-			store, err := consul.NewClient(opts)
-			if err != nil {
-				logger.V(0).Error(err, "failed to connect to consul")
+			// reference for using another persistence layer
+			/*
+				opts := consul.Options{
+					Scheme:  "http",
+					Address: "localhost:8500",
+					Codec:   encoding.JSON,
+				}
+				store, err := consul.NewClient(opts)
+				if err != nil {
+					logger.V(0).Error(err, "failed to connect to consul")
+					os.Exit(1)
+				}
+				var repo repository.Actions
+				repo = &persistence.GoKV{Store: store, Ctx: ctx}
+			*/
+			// add grpcsvr.WithPersistence(repo) to grpc.RunServer
```

Comment:
delete?

Reply:
ah yes, will remove. thanks

Reply:
actually, this is being removed in a different PR.

Reply:
heh :D, I thought it looked familiar. Lets ignore it here then.